app/javascript/react/containers/AmusementParksShowContainer.js

```diff
@@ -1,38 +1,78 @@
 import React from 'react';
-import ParkInfoTile from '../components/ParkInfoTile';
+
+import ParkShowTile from '../components/ParkShowTile'
+import ReviewsContainer from './ReviewsContainer'
+import ReviewFormContainer from './ReviewFormContainer'
 
 class AmusementParksShowContainer extends React.Component {
   constructor(props){
     super(props);
     this.state = {
-      amusementPark: {}
+      amusementPark: {},
+      reviews: []
     };
+
+    this.addReview = this.addReview.bind(this)
   }
 
   componentDidMount(){
-    fetch(`/api/v1/amusement_parks/${this.props.params.id}`)
+    fetch(`/api/v1/amusement_parks/${this.props.params.id}.json`)
     .then(response => {
       if (response.ok) {
         return response;
       } else {
-        let errrorMessage = `${response.status} (${response.statusText})`,
+        let errorMessage = `${response.status} (${response.statusText})`,
           error = new Error(errorMessage);
         throw(error);
       }
     })
     .then(response => response.json())
-    .then(body => { this.setState({
-        amusementPark: body.amusement_park
+    .then(body => {
+      this.setState({
+        amusementPark: body.amusement_park,
+        reviews: body.reviews
       })
     })
     .catch(error => console.error(`Error in fetch: ${error.message}`));
   }
 
+  addReview(payload){
+    fetch(`/api/v1/amusement_parks/${this.props.params.id}/reviews.json`, {
+      credentials: 'same-origin',
+      headers: { 'Content-Type': 'application/json',
+      'X-Requested-With': 'XHMLttpRequest' },
+      method: 'POST',
+      body: JSON.stringify(payload)
+    })
+      .then(response => {
+        if(response.ok){
+          return response
+        } else {
+          let errorMessage = `${response.status} (${response.statusText})`,
+              error = new Error(errorMessage)
+          throw(error)
+        }
+      })
+      .then(response => response.json())
+      .then(body => {
+        if(body.review){
+          this.setState({ reviews: this.state.reviews.concat(body.review) })
```

Comment:
good. `concat` is good for immutability because it does not mutate the existing array.